src/vs/code/electron-main/window.ts

```diff
@@ -32,6 +32,7 @@ export interface IWindowCreationOptions {
 	extensionDevelopmentPath?: string;
 	allowFullscreen?: boolean;
 	titleBarStyle?: 'native' | 'custom';
+	vscodeWindowId?: string;
```

Comment:
@Tyriar I would suggest to call this `id` and instead fix all users of the current `id` property to use `vscodeWindows.win.id` because that makes it very clear what ID is being asked.